File: simple_examples.py
import argparse

# function to add two numbers
def add(a, b):
    return a + b


# function to add or multiply two numbers based on an operator argument
def add_or_multiply(a, b, operator):
    if operator == "add":
        return a + b
    elif operator == "multiply":
        return a * b
    else:
        raise ValueError("Invalid operator")

# setup main
if __name__ == "__main__":
    # parse arguments
    parser = argparse.ArgumentParser()
    parser.add_argument("--a", type=int, default=1)
    parser.add_argument("--b", type=int, default=2)
    parser.add_argument("--operator", type=str, default="add")
    args = parser.parse_args()

    # call add_or_multiply
    result = add_or_multiply(args.a, args.b, args.operator)

    # print result
    print(result)

File: simple_examples.c
/* function to add two numbers */
int add(int a, int b)
{
    return a + b;
}

/* function to add, subtract, multiply and divide two numbers based on an operator */
int calculate(int a, int b, char op)
{
    int result = 0;
    switch (op)
    {
        case '+':
            result = add(a, b);
            break;
        case '-':
            result = subtract(a, b);  /* Commentary: not sure where subtract is defined */
            break;
        case '*':
            result = multiply(a, b);  /* Commentary: not sure where multiply is defined */
            break;
        case '/':
            result = divide(a, b);  /* Commentary: not sure where divide is defined */
            break;
        default:
            printf("Invalid operator");
    }
    return result;
}

/* main function */
int main()
{
    int a, b, result;
    char op;
    printf("Enter two numbers: ");
    scanf("%d %d", &a, &b);
    printf("Enter an operator: ");
    scanf(" %c", &op);
    result = calculate(a, b, op);
    printf("Result: %d", result);
    return 0;
}

